handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+import { useState } from 'react';
+
+const NewStudentForm = () => {
+
+    const [formFields, setFormFields] = useState({
+        name: '',
+        email: ''
+    });
+
+    const onNameChange = (event) => {
+        setFormFields({
+            ...formFields,
+            name: event.target.value
+        })
+    };
+
+    const onEmailChange = (event) => {
+        setFormFields({
+            ...formFields,
+            email: event.target.value
+        })
+    };
+
+    return (
+        <form>
+            <div>
+                <label htmlFor="fullName">Name:</label>
+                <input
+                    name="fullName"
+                    value={formFields.name}
+                    onChange={onNameChange}/>
+            </div>
+            <div>
+                <label htmlFor="email">Email:</label>
+                <input name="email"
+                    value={formFields.email}
+                    onChange={onEmailChange}/>
+            </div>
+            <input
+                type="submit"
+                value="Add Student"
+            />
+        </form>
+    );
+}
+
+export default NewStudentForm;
+```
+<!-- prettier-ignore-end -->
+
+</details>
+
+The `NewStudentForm` component is all well and good. It follows the controlled component pattern, so its state is in sync with the input fields.
+
+However, all student data is managed in the `App` component. Our new student form doesn't actually add a student!
+
+To handle form submissions, and bring the data to the `App` component, we need to lift state up.
+
+## Sofia's Plan
+
+Passing down event handlers and lifting state up follows exact same process that she's done before in React development. The _only exception_ is she needs to learn how to prevent default form behavior.
+
+Sofia's goal is to get the data from the `NewStudentForm` up to the `App` component.
+
+The `App` component is already the single source of truth for her student data, so the remaining steps of her plan are:
+
+1. Pass down event handlers from `App` to the `NewStudentForm`
+1. Configure the form submission in `NewStudentForm` to lift state up
+
+## Passing Down New Student Handlers
+
+Sofia starts by making a method in `App` that adds a new student to the student data:
+
+<!-- prettier-ignore-start -->
+```js
+    const addStudentData = newStudent => {
+        // Duplicate the student list
+        const newStudentList = [...studentData];
+
+        // Logic to generate the next valid student ID
+        const nextId = Math.max(...newStudentList.map(student => student.id)) + 1;
+
+        newStudentList.push({
+            id: nextId,
+            nameData: newStudent.nameData,
+            emailData: newStudent.emailData,
+            isPresentData: false,
+        });
+
+        setStudentData(newStudentList);
+    };
+```
+<!-- prettier-ignore-end -->
+
+There are no rules about this `addStudentData` function, and Sofia could have written this logic using any approach or code style!
```

Comment:
I can't bring myself to say there are _no_ rules, but no more than we usually have. just so long as it meets her project needs
```suggestion
There are no special rules about this `addStudentData` function. Sofia could have written this logic using any approach or code style that meets the needs of her project!
```